cfgov/v1/models/banners.py

```diff
@@ -0,0 +1,38 @@
+from django.core.validators import RegexValidator
+from django.db import models
+from django.utils.safestring import mark_safe
+
+from wagtail.admin.edit_handlers import FieldPanel, StreamFieldPanel
+from wagtail.core.fields import StreamField
+
+from v1.atomic_elements.molecules import Notification
+
+
+class Banner(models.Model):
+    title = models.CharField(
+        max_length=255,
+        help_text='For internal reference only; does not appear on the site.'
+    )
+    url_patterns = models.CharField(
+        max_length=1000,
+        verbose_name='URL patterns',
+        help_text=mark_safe('A regular expression pattern for matching URLs '
+                            'that should show the banner, for example: '
+                            '<code>contact-us|^/complaint/$</code>'),
+        validators=[
+            RegexValidator(regex=r'[A-Za-z0-9\-_.:/?&|\^$]')
+        ],
+    )
+    # TODO: add `min_num` and `max_num` arguments of 1 to the StreamField below
+    # if https://github.com/wagtail/wagtail/pull/5185 ever gets merged.
```

Comment:
You can do this with a custom `StreamBlock`, for example as we do [here](https://github.com/cfpb/cfgov-refresh/blob/1b478eb9fe2b61ea29a344e898028da4f92130eb/cfgov/v1/models/sublanding_filterable_page.py#L29), but unfortunately it only works on the backend, at validation time, and won't prevent you from adding additional blocks in the editor. Wagtail 5185 has this same limitation, though.

For example:

```py
class BannerContent(StreamBlock):
    content = molecules.Notification()

    class Meta:
        block_counts = {
            'content': {'max_num': 1},
        }
```

Reply:
Yeah, it just didn't seem worth adding the extra UI complication (adding the StreamBlock, then adding the Notification).

Reply:
This doesn't add any extra UI complication -- the Wagtail editing experience is the same. I missed one piece in my suggestion, which is the way to define the field:

```py
class Banner(models.Model):
    content = StreamField(BannerContent)
```

This avoids needing to add another top-level block.

Reply:
Ah, good call.

Reply:
Amended original commit with this change, now in fc9e180.